src/VaultAuthClient.php

```diff
@@ -0,0 +1,62 @@
+<?php
+namespace SublimeSkinz\SublimeVault;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+use SublimeSkinz\SublimeVault\AwsClient;
+use SublimeSkinz\SublimeVault\VaultLogger;
+
+class VaultAuthClient
+{
+
+    protected $awsClient;
+    protected $logger;
+    protected $guzzleClient;
+
+    public function __construct()
+    {
+        $this->awsClient = new AwsClient();
+        $this->guzzleClient = new Client();
+        $this->logger = new VaultLogger();
+    }
+
+    /**
+     * Get Vault token from appRole credentials
+     * 
+     * @param string $addr
+     * @param string $bucketName
+     * @param string $credsPath
+     * @return string|null
+     */
+    public function getVaultTokenWithAppRole($addr, $bucketName, $credsPath)
+    {
+        if ($bucketName && $credsPath) {
+            $creds = $this->awsClient->fetchAppRoleCreds($bucketName, $credsPath);
+            if (is_null($creds)) {
+                return null;
+            }
+            try {
+                $options = [
```

Comment:
also here the options variable is not very useful you can set it directly in the http client